docs/sources/flow/reference/components/loki.process.md

```diff
@@ -221,6 +294,236 @@ stage {
 }
 ```
 
+### regex block
+
+The `regex` inner block configures a processing stage that parses log lines
+using regular expressions and uses named capture groups for adding data into
+the shared extracted map of values.
+
+The following arguments are supported:
+
+Name          | Type      | Description                                                         | Default | Required
+------------- | --------- | ------------------------------------------------------------------- | ------- | --------
+`expression`  | `string`  | A valid RE2 regular expression. Each capture group must be named.   |         | yes
+`source`      | `string`  | Name from extracted data to parse. If empty, uses the log message.  | `""`    | no
+
+
+The `expression` field needs to be a Go RE2 regex string. Every capture group (re) is set into the extracted map, so it must be named like: `(?P<name>re)`. The name of the capture group is then used as the key in the extracted map.
+
+<!--
+We don't care about YAML, what does River do instead???
+
+Because of how YAML treats backslashes in double-quoted strings, note that all backslashes in a regex expression must be escaped when using double quotes. For example, all of these are valid:
+
+expression: \w*
+expression: '\w*'
+expression: "\\w*"
+But these are not:
+
+expression: \\w* (only escape backslashes when using double quotes)
+expression: '\\w*' (only escape backslashes when using double quotes)
+expression: "\w*" (backslash must be escaped)
+-->
+
+If the `source` is empty, then the stage uses attempts to parse the log line
+itself.
+
+Given the following log line and regex stage, the extracted values are:
+
+```
+2019-01-01T01:00:00.000000001Z stderr P i'm a log message!
+stage {
+  regex {
+    expression = "^(?s)(?P<time>\\S+?) (?P<stream>stdout|stderr) (?P<flags>\\S+?) (?P<content>.*)$"
+  }
+}
+
+
+time: 2019-01-01T01:00:00.000000001Z,
+stream: stderr,
+flags: P,
+content: i'm a log message
+```
+
+On the other hand, if the `source` value is set, then the regex is applied to
+the value stored in the shared map under that name.
+
+Let's see what happens when the following log line is put through this
+two-stage pipeline:
+```
+{"timestamp":"2022-01-01T01:00:00.000000001Z"}
+
+stage {
+  json {
+    expressions = { time = "timestamp" }
+  }
+}
+stage {
+  regex {
+    expression = "^(?P<year>\\d+)"
+    source     = "time"
+  }
+}
+```
+
+The first stage would add the following key-value pairs into the extracted map:
+```
+time: 2022-01-01T01:00:00.000000001Z
+```
+
+Then, the regex stage would parse the value for time from the shared values and
+append the following key-value pairs back into the extracted values map:
```

Comment:
```suggestion
Then, the regex stage parses the value for time from the shared values and
appends the subsequent key-value pairs back into the extracted values map:
```

Reply:
Yes, fixed it for the previous instance as well!